packages/js/components/src/split-dropdown/split-dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { cloneElement, createElement } from '@wordpress/element';
+import { Icon, chevronDown, chevronUp } from '@wordpress/icons';
+import { Button, ButtonGroup, Dropdown } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+export type SplitDropdownProps = {
+	className?: string;
+	disabled?: boolean;
+	variant?: string;
+	children: JSX.Element[];
+};
+
+export const SplitDropdown: React.FC< SplitDropdownProps > = ( {
+	className = '',
+	children,
+	disabled,
+	variant = 'primary',
+}: SplitDropdownProps ) => {
+	const groupActionProps = Object.assign(
+		{ variant },
+		disabled ? { disabled } : {}
+	);
+	const mainActionProps = {
+		...groupActionProps,
+		className: `woocommerce-split-dropdown__main-button ${ className }`,
+	};
+	const [ mainAction, ...menuActions ] = children;
+	const menuIcon = chevronDown;
+	const menuIconExpanded = chevronUp;
+	return (
+		<ButtonGroup className={ `woocommerce-split-dropdown ${ className }` }>
```

Comment:
The `classnames` package might be a slightly cleaner way of handling this.
```js
classnames( 'woocommerce-split-dropdown', className )
```
And then if `className` is `null` it will not be shown, nor any spaces.

Reply:
I will look into that, thanks!